tests/test_data/test_shear.py

```diff
@@ -0,0 +1,191 @@
+    results_sheared = shear_module(
+        copy.deepcopy(results), random_negative_prob=0.)
+    results_gt = copy.deepcopy(results)
+    img_s = np.array([[1, 2, 3, 4], [0, 5, 6, 7]], dtype=np.uint8)
+    img_s = np.stack([img_s, img_s, img_s], axis=-1)
+    img_s[1, 0, :] = np.array(img_fill_val)
+    results_gt['img'] = img_s
+    results_gt['gt_bboxes'] = np.array([[0., 0., 3., 1.]], dtype=np.float32)
+    results_gt['gt_bboxes_ignore'] = np.array([[2., 0., 4., 1.]],
+                                              dtype=np.float32)
+    gt_masks = np.array([[0, 1, 1, 0], [0, 0, 1, 0]],
+                        dtype=np.uint8)[None, :, :]
+    results_gt['gt_masks'] = BitmapMasks(gt_masks, 2, 4)
+    results_gt['gt_semantic_seg'] = np.array(
+        [[1, 2, 3, 4], [255, 5, 6, 7]], dtype=results['gt_semantic_seg'].dtype)
+    check_shear(results_gt, results_sheared)
+
+    # test PolygonMasks with shear horizontally, magnitude=1
+    results = construct_toy_data(poly2mask=False)
+    results_sheared = shear_module(
+        copy.deepcopy(results), random_negative_prob=0.)
+    gt_masks = [[np.array([1, 0, 2, 0, 3, 1, 2, 1], dtype=np.float)]]
+    results_gt['gt_masks'] = PolygonMasks(gt_masks, 2, 4)
+    check_shear(results_gt, results_sheared)
+
+    # test shear vertically, magnitude=-1
+    img_fill_val = 128
+    results = construct_toy_data()
+    transform = dict(
+        type='Shear',
+        level=10,
+        prob=1.,
+        img_fill_val=img_fill_val,
+        direction='vertical',
+        max_shear_magnitude=1.)
+    shear_module = build_from_cfg(transform, PIPELINES)
+    results_sheared = shear_module(
+        copy.deepcopy(results), random_negative_prob=1.)
+    results_gt = copy.deepcopy(results)
+    img_s = np.array([[1, 6, img_fill_val, img_fill_val],
+                      [5, img_fill_val, img_fill_val, img_fill_val]],
+                     dtype=np.uint8)
+    img_s = np.stack([img_s, img_s, img_s], axis=-1)
+    results_gt['img'] = img_s
+    results_gt['gt_bboxes'] = np.empty((0, 4), dtype=np.float32)
+    results_gt['gt_labels'] = np.empty((0, ), dtype=np.int64)
+    results_gt['gt_bboxes_ignore'] = np.empty((0, 4), dtype=np.float32)
+    gt_masks = np.array([[0, 1, 0, 0], [0, 0, 0, 0]],
+                        dtype=np.uint8)[None, :, :]
+    results_gt['gt_masks'] = BitmapMasks(gt_masks, 2, 4)
+    results_gt['gt_semantic_seg'] = np.array(
+        [[1, 6, 255, 255], [5, 255, 255, 255]],
+        dtype=results['gt_semantic_seg'].dtype)
+    check_shear(results_gt, results_sheared)
+
+    # test PolygonMasks with shear vertically, magnitude=-1
+    results = construct_toy_data(poly2mask=False)
+    results_sheared = shear_module(
+        copy.deepcopy(results), random_negative_prob=1.)
+    gt_masks = [[np.array([1, 0, 2, 0, 2, 0, 1, 0], dtype=np.float)]]
+    results_gt['gt_masks'] = PolygonMasks(gt_masks, 2, 4)
+    check_shear(results_gt, results_sheared)
```

Comment:
For a similar mask, the results after shear augmentation on PolygonMasks and BitMapMasks should be the same, we could also check this case.